### ISSUE TITLE
vscode.dev lsp fails time to time

### ISSUE THREAD

**Issue by heejaechang** [microsoft/pylance-release#2743]
type

import os

and then go to definitino on os

once ```__init__.pyi``` is opened, move around to see hover info while the file is still in the middle of processing.

bunch of requests fails

```
[Error - 1:57:09 PM] Request textDocument/foldingRange failed.
  Message: Request textDocument/foldingRange failed with message: request cancelled
  Code: -32603 
[Error - 1:59:41 PM] Request textDocument/hover failed.
  Message: Request textDocument/hover failed with message: request cancelled
  Code: -32603 
[Error - 1:59:41 PM] Request textDocument/hover failed.
  Message: Request textDocument/hover failed with message: request cancelled
  Code: -32603 
[Error - 1:59:41 PM] Request textDocument/hover failed.
  Message: Request textDocument/hover failed with message: request cancelled
  Code: -32603 
```

need to figure out why it is happening

**Comment by heejaechang** [comment]
it looks like cancellation becomes request failure in browser. it works as expected in nodejs/vscode.

**Comment by luabud** [comment]
Is https://github.com/microsoft/pylance-release/issues/2888 a dupe of this?

**Comment by debonte** [comment]
> Is #2888 a dupe of this?

No, that's something different. Rather than `request cancelled` it says `Cannot read properties of undefined (reading 'startOffset')`

**Comment by debonte** [comment]
This issue has been fixed in version 2022.6.10, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#2022610-8-June-2022)
